File: src/features/storage/domain/settingsRepository.ts
/**
 * Интерфейс key-value хранилища настроек.
 *
 * Ключи: language, tts_voice, tts_rate, tts_volume и др.
 */

export interface SettingsRepository {
  getString(key: string): Promise<string | null>;
  setString(key: string, value: string): Promise<void>;

  getNumber(key: string): Promise<number | null>;
  setNumber(key: string, value: number): Promise<void>;

  getBool(key: string): Promise<boolean | null>;
  setBool(key: string, value: boolean): Promise<void>;

  /** Удалить значение */
  remove(key: string): Promise<void>;

  /** Очистить все настройки */
  clear(): Promise<void>;
}

File: lib/features/storage/domain/settings_repository.dart
/// Репозиторий пользовательских настроек (§4.5 ТЗ).
///
/// Key-value хранилище в SQLite. Настройки:
/// - Язык интерфейса / ASR ('ru' / 'en')
/// - Голос TTS
/// - Скорость речи (0.5–2.0)
/// - Feature flags (перезаписываемые)
abstract class SettingsRepository {
  /// Получает строковую настройку по ключу.
  Future<String?> getString(String key);

  /// Сохраняет строковую настройку.
  Future<void> setString(String key, String value);

  /// Получает числовую настройку.
  Future<double?> getDouble(String key);

  /// Сохраняет числовую настройку.
  Future<void> setDouble(String key, double value);

  /// Получает булеву настройку.
  Future<bool> getBool(String key, {bool defaultValue = false});

  /// Сохраняет булеву настройку.
  Future<void> setBool(String key, bool value);

  /// Ключи настроек — для единообразия.
  static const String keyLanguage = 'language';
  static const String keyTtsVoice = 'tts_voice';
  static const String keyTtsRate = 'tts_rate';
  static const String keyTtsVolume = 'tts_volume';
  static const String keyOnboardingComplete = 'onboarding_complete';
}
